docs/samples/Microsoft.ML.Samples/Trainers.cs

```diff
@@ -2,34 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Microsoft.ML.Runtime.Data;
-using Microsoft.ML.Runtime.Learners;
-using Microsoft.ML.Trainers;
-using System;
+// the alignment of the usings with the methods is intentional so they can display on the same level in the docs site. 
+        using Microsoft.ML.Runtime.Data;
+        using Microsoft.ML.Runtime.Learners;
+        using Microsoft.ML.Trainers;
+        using System;
 
+// NOTE: WHEN ADDING TO THE FILE, ALWAYS APPEND TO THE END OF IT. 
```

Comment:
>ALWAYS APPEND TO THE END OF IT. [](start = 34, length = 31)

what will happen if I need to add another using?

Reply:
//If you change the existing content, check that the files referencing it in the XML documentation are still correct, as they reference
// line by line.


see the line numbers, after the range query string param.
[!code-csharp[SDCA](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Trainers.cs?range=6-10,19-76 "The SDCA regression example.")]

not ideal, i know. 
Better than having examples in the xml, that never compile, better than not having any examples in the documentation. 
And if it is start to feeling like too much, we can have one example per file, to minimize maintenance. 

---
In reply to: [224617093](https://github.com/dotnet/machinelearning/pull/1189#discussion_r224617093) [](ancestors = 224617093)

Reply:
My 2 cents:
One example per file makes a lot of sense if the filename is very concise.

Reply:
the problem is actually with the lines referenced. Let's see how it progresses before changing it. 

---
In reply to: [224621387](https://github.com/dotnet/machinelearning/pull/1189#discussion_r224621387) [](ancestors = 224621387)